localstack-core/localstack/aws/connect.py

```diff
@@ -660,7 +660,11 @@ def resolve_dns_from_upstream(hostname: str) -> str:
     if len(response.answer) == 0:
         raise ValueError(f"No DNS response found for hostname '{hostname}'")
 
-    ip_addresses = list(response.answer[0].items.keys())
+    ip_addresses = []
+    for answer in response.answer:
+        if answer.match(dns.rdataclass.IN, dns.rdatatype.A, dns.rdatatype.TYPE0):
```

Comment:
q: what is `TYPE0` in this context? The argument is called `covers` but I don't know what that means.

Does this mean "anything" as `0` is not a special value perhaps?

Reply:
Good question. 😬 I hoped you'd be able to answer this, I've printed all the different `answer` objects and printed the `covers` field of them, and all of them returned `TYPE0`. Looking more into it, it seems related with `RRSIG` record types: https://simpledns.plus/help/rrsig-records

https://dnspython.readthedocs.io/en/latest/rdata-set-classes.html#dns.node.Node.find_rdataset

> covers, a dns.rdatatype.RdataType, the covered type. Usually this value is dns.rdatatype.NONE, but if the rdtype is dns.rdatatype.SIG or dns.rdatatype.RRSIG, then the covers value will be the rdata type the SIG/RRSIG covers. The library treats the SIG and RRSIG types as if they were a family of types, e.g. RRSIG(A), RRSIG(NS), RRSIG(SOA). This makes RRSIGs much easier to work with than if RRSIGs covering different rdata types were aggregated into a single RRSIG rdataset.

So I suppose it does not apply to type `A` records and can safely be put to `TYPE0` or `NONE` (both are equal to `0` in the Enum, so I've replaced `TYPE0` by `NONE` which is a bit more explicit?